### PR TITLE
Added unit tests

### PR DESCRIPTION
**Pull request by jspada**
* Added unit test vectors to be shared with other implementations
    * Added Makefile and split code into reference_signer and unit_tests
    * Made unit tests run on build
    * Updated documentation
    * Implemented get_address()
    * Moved read_public_key_compressed() and prepare_memo() to crypto.c
    * Added verbose mode to unit tests
    * Added mode to speedup and automate generation of Ledger unit
      tests and reduce scope for errors
    * Added gitignore

### CODE REVIEW

File: crypto.c [hunk 1]
```diff
@@ -27,13 +27,15 @@
 
 #include <assert.h>
 #include <inttypes.h>
+#include <openssl/sha.h>
```
Comment (imeckler) [inline crypto.c]: It would be nice not to depend on external libraries. What is this used for?

Comment (jspada) [reply crypto.c]: Mina address generation.  I will switch to an in-built one to be more portable.

File: crypto.c [hunk 2]
```diff
@@ -117,9 +118,20 @@ unsigned int field_eq(const Field a, const Field b)
...
+void scalar_copy(Scalar b, const Scalar a)
 {
-    fiat_pasta_fq_copy(c, a);
+    fiat_pasta_fq_copy(b, a);
+}
+
+void scalar_from_words(Scalar a, const uint64_t words[4])
+{
+    uint8_t *p = (uint8_t *)words;
+    uint8_t bytes[sizeof(Scalar)];
```
Comment (imeckler) [inline crypto.c]: This function relies on word internal endianness right? It looks like it's not really needed -- AFAICT you really want scalar_from_bytes_le or something?

Can you change it to not rely on word-internal endianness? I know this is only used in unit tests so it is fine now, but it's important to keep this repo endianness-independent.

File: curve_checks.c [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+//
+//    Imported from https://github.com/MinaProtocol/c-reference-signer/blob/master/curve_checks.c
+
+#include "curve_checks.h"
+
+bool curve_checks(void)
+{
+    Affine a3;
+    Affine a4;
+    union {
```
Comment (imeckler) [inline curve_checks.c:11]: a mix of :grimacing: and :joy:

**Review by imeckler — CHANGES_REQUESTED** [review]

**Review by imeckler — APPROVED** [review]

File: curve_checks.c (hunk 3 above)
Comment (jspada) [reply curve_checks.c:11]: Yeah - this was the first time I've used union in a long time.  Simply put, we were out of stack space and I had no choice! :)

File: crypto.c (hunk 2 above)
Comment (jspada) [reply crypto.c]: Fixed!

**Pull request merged**